src/v/serde/json/parser.cc

```diff
@@ -0,0 +1,570 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "serde/json/parser.h"
+
+#include "base/unreachable.h"
+#include "container/fragmented_vector.h"
+#include "serde/json/detail/numeric.h"
+#include "serde/json/detail/string.h"
+
+#include <seastar/coroutine/maybe_yield.hh>
+
+#include <cstddef>
+#include <cstdint>
+#include <variant>
+
+#define ENABLE_TRACE 0
+
+#if ENABLE_TRACE
+#define TRACE(...)                                                             \
+    do {                                                                       \
+        fmt::print(__VA_ARGS__);                                               \
+    } while (false)
+#else
+#define TRACE(...)                                                             \
+    do {                                                                       \
+    } while (false)
+#endif
+
+namespace experimental::serde::json {
+
+std::ostream& operator<<(std::ostream& os, token t) {
```

Comment:
nit: use as_format function instead?

Reply:
What is `as_format`? Grepping for `as_format` in our codebase doesn't lead to anything.

Reply:
https://fmt.dev/11.1/api/#udt


The version we are using only supports as_format for enums, the next major version added support for "regular types too".

Reply:
It's the simpler less verbose way of making your type formattable without fmt::formatter